diff --git a/HubAccelerator/csvUpdater.py b/HubAccelerator/csvUpdater.py
index 996fca3..f9341a7 100755
--- a/HubAccelerator/csvUpdater.py
+++ b/HubAccelerator/csvUpdater.py
@@ -1,4 +1,4 @@
-#!/usr/local/bin/python3
+#!/opt/homebrew/bin/python3
 """
 Update Security Hub findings en bloc/en masse from a CSV file
 
@@ -62,18 +62,34 @@ class InputDiscriminator:
 ################################################################################
 #### 
 ################################################################################
-def choose (default=None, *choices):
+def choose (choices={}):
     """
     Choose between an option and an environment variable (the option always
     has priority, if specified)
     """
-    answer = default
+    for name, value in choices.items():
+        _LOGGER.info(f'csvUpdate.494010i choice {name}: {value}')
 
-    for choice in choices:
-        if choice:
-            answer = choice
+        if value:
+            _LOGGER.info(f'csvUpdate.494012 using choice {name}')
+            answer = value
             break
 
+    if not isinstance(answer, list):
+        candidate = answer
+
+        try:
+            answer = json.loads(candidate)
+        except Exception as thrown:
+            try:
+                answer = re.sub('\s*[\[\]\s]\s*', '', candidate).split(',')
+            except Exception as thrown:
+                _LOGGER.error(f'494013e cannot parse {candidate} at all: {thrown}')
+                raise thrown
+
+        _LOGGER.info(f'494016i candidate {candidate} ({type(candidate).__name__}) ' +
+            f'to {answer} ({type(answer).__name__})')
+
     return answer
 ################################################################################
 #### Invocation-independent process handler
@@ -91,11 +107,11 @@ def executor (role=None, region=None, debug=False, input=None):
     ssmActor = csvo.SsmActor(role=role, region=region)
 
     # Get a list of Security Hub regions we wish to act on
-    regions = choose(
-        os.environ.get("CSV_SECURITYHUB_REGIONLIST"),
-        re.compile("\s*,\s*").split(getattr(ssmActor, "/csvManager/regionList")),
-        ssmActor.getSupportedRegions(service="securityhub")
-    )
+    regions = choose({
+        'Environment variable CSV_SECURITYHUB_REGIONLIST': os.environ.get("CSV_SECURITYHUB_REGIONLIST"),
+        'SSM region list /csvManager/regionList': re.compile("\s*,\s*").split(getattr(ssmActor, "/csvManager/regionList", region)),
+        'ssm:getSupportedRegions API': ssmActor.getSupportedRegions(service="securityhub")
+    })
     
     _LOGGER.info("494010i selected SecurityHub regions %s" % regions)
 
@@ -244,7 +260,7 @@ def lambdaHandler ( event = None, context = None ):
         roleArn = event.get("roleArn")
         input = event.get("input")
         debug = event.get("debug")
-        region = event.get("primaryRegion")
+        region = event.get("region")
 
         # Do the work
         answer = executor(
