diff --git a/HubAccelerator/csvObjects.py b/HubAccelerator/csvObjects.py
index 83f3ef0..3151887 100755
--- a/HubAccelerator/csvObjects.py
+++ b/HubAccelerator/csvObjects.py
@@ -711,7 +711,7 @@ class ActorException (Exception):
 ################################################################################
 class Actor:
     _REGION_MODE_SINGLE = 1    # A simple, single-region client
-    _REGION_MODE_MULTIPLE = 1   # Requires a ServiceRegionBroker
+    _REGION_MODE_MULTIPLE = 2   # Requires a ServiceRegionBroker
     """
     An abstract class for API functions. The class defines clients in each of 
     a set of supported regions, and then carries out actions using those
@@ -735,6 +735,8 @@ class Actor:
         self.principal = None
         self.service = service
 
+        _LOGGER.debug(f'496135d {service} {region} {type(region)} {role}')
+
         # Some things depend on whether we were passed a region or a list 
         if isinstance(region, list):
             self.regions = region
@@ -746,6 +748,8 @@ class Actor:
             self.regions = [ region ]
             self.mode = Actor._REGION_MODE_SINGLE
 
+            _LOGGER.debug(f'496145d service {service} set to _REGION_MODE_SINGLE')
+
         else:
             raise ActorException("496150t region must be a region name or list of regions [%s]" % region)
 
@@ -834,13 +838,17 @@ class Actor:
         sts:assume_role to gain the privileges associated with the supplied
         role ARN.
         """
-        _LOGGER.debug("496200d request to authorize %s client region %s"
-            % (self.service, regions[0]))
+        _LOGGER.debug(f'496203d regions {regions} {type(regions)} {regions[0]}')
+        region = regions[0] if isinstance(regions, list) \
+            else regions if isinstance(regions, str) else None;
+
+        _LOGGER.debug(f'496200d request to authorize {self.service}' + \
+            f' client region {region}')
 
         # Obtain an STS client
         try:
             # Obtain an STS client
-            client = boto3.client("sts", region_name=regions[0])
+            client = boto3.client("sts", region_name=region)
             _LOGGER.debug("496210d obtained STS client %s" % client)
 
             # No role supplied - use environment credentials
